### ISSUE TITLE
RecurringJobManager doesn't respect queues

### ISSUE THREAD

**Issue by Z1ni** [HangfireIO/Hangfire#926]
I have a situation where there are multiple Hangfire job servers using one database. Each of the servers have unique name and unique queue name that they use to queue jobs. Each of the servers also has its own recurring jobs. Everything else works fine but I've noticed that Hangfire is trying to execute recurring jobs on wrong job servers that aren't configured to use the queues.

E.g. I have two job servers, A and B.
A is configured to use queue "queueA" and B is configured to use queue "queueB". I queue a recurring job in server A in queue "queueA" to run every 5 minutes. Every 5 minutes job server A runs the job, but also server B tries to run the job even though it hasn't been configured to do so.

This seems like a bug to me.

**Comment by Z1ni** [comment]
Seems that this is a duplicate of #908. I will close this if the issue gets some activity.

**Comment by mikechamberlain** [comment]
Yeah there seems to be something screwed up with queues and multiple servers in HF. Things seems to work OK on my local machine but break when I move my code to our production cluster.

**Comment by mikanyg** [comment]
I have seen this behaviour as well, and really bloated our logs, untill filtering them out. But surely this seems like a wrong behavior. How else could one partition job execution in a micro-service architecture with a single monitoring dashboard?

**Comment by mikanyg** [comment]
@odinserj Any thoughts on how to have HF behave well in a partitioned multi server setup? I'm considering looking into doing a PR to fix it, if possible. But would like your thoughts on the issue and possible fix.

**Comment by moskvitin** [comment]
@odinserj is there an official opinion about this behavior?

**Comment by timavaza** [comment]
Same happening for us. Seems a clear bug. I'm wondering, if the execution Class for queue A's task exists on server for queue B, with the server go ahead and process the task from the wrong queue?

**Comment by chris1411** [comment]
This is a big issue for my company as well. Its causing many of our jobs to be delayed unnecessarily since they only have a 50/50 chance of hitting the correct queue. Any updates on this would be appreciated.

**Comment by Finestmedia** [comment]
Big issue for us aswell. Same scenario as OP. First time the job is run by correct server/worker (separate queues), but when it fails and is retried, half of the time it ends up trying to run on the wrong worker. We would like to keep Hangfire on one database and one dashboard. Since it's such an old case, I have little hope it will get any other solution at all.

**Comment by Codearella** [comment]
Any update on this issue?

**Comment by Compl1cation** [comment]
I would have loved to use queues but the randomness of this issue just risks the job scheduler throwing and putting the job in permanent not enqueued state.
The only thing that solved this problem for me was using a separate database per application.

**Comment by ZlobnyiSerg** [comment]
@odinserj could you please consider to include this fix in some nearest releases? Very unexpected behaviour. Faced couple of days ago in production after more than 3 years with Hangfire.

**Comment by ZlobnyiSerg** [comment]
I think also https://github.com/HangfireIO/Hangfire/issues/595 is duplicate of this

**Comment by berdem** [comment]
@odinserj we at least would know if there is any plan to fix this behavior, please ?

**Comment by alefcarlos** [comment]
Same here :(

**Comment by smashpl** [comment]
And the same problem for me :(

**Comment by GeXiaoguo** [comment]
I have [this solution](https://github.com/GeXiaoguo/Hangfire.MAMQSqlExtension) running with hyper care for a few months now with ~25 jobs in 4 different queues and 2 servers not sharing code base. Seems to be working fine up to now.

**Comment by alefcarlos** [comment]
@GeXiaoguo I'm gonna try it! Thank you! Today I need to update job library references in all Servers, but this is easy to forget.

**Comment by DaleCam** [comment]
I believe I was able to fix this by using a different schema for each app. You can specifiy this with the setting below:
```            
builder.Services.AddHangfire(x => x.UseSqlServerStorage(builder.Configuration.GetConnectionString("HangfireJobs"), new SqlServerStorageOptions()
            {
                ...your settings here ...
                SchemaName = "YourSchemaName"
            }));
```
